bin/module-update/module-update.sh

```diff
@@ -0,0 +1,184 @@
+
+  hcledit attribute rm module.infra.route53_hosted_zone_name -u -f "${INFRA_DIR}/main.tf"
+  hcledit attribute rm module.infra.route53_hosted_zone_private -u -f "${INFRA_DIR}/main.tf"
+
+  hcledit attribute rm route53_hosted_zone_name -u -f "${INFRA_VARS}"
+
+}
+
+set_vpc_cni_import() {
+
+  local nodes_main_tf_path="${NODES_DIR}/imports.tf"
+
+  if hcledit block get "import" -f "$nodes_main_tf_path" | grep -q 'to = module.nodes.aws_eks_addon.pre_compute_addons\["vpc-cni"\]'; then
+    printf 'Import block already exists for module.nodes.aws_eks_addon.pre_compute_addons["vpc-cni"]. \n'
+    return
+  fi
+
+  printf "Creating import block for 'vpc-cni' EKS addon. After this configuration is applied and the vpc-cni addon is imported, the %s file can be removed\n" "$nodes_main_tf_path"
+
+  cat <<-EOB >>"$nodes_main_tf_path"
+import {
+  to = module.nodes.aws_eks_addon.pre_compute_addons["vpc-cni"]
+  id = "\${local.eks.cluster.specs.name}:vpc-cni"
+}
+EOB
+
+  echo
+}
+
+update_mod_source() {
+
+  local mod_dir="$1"
+  local mod_version="$2"
+
+  local update_mod_sh_path="${mod_dir}/set-mod-version.sh"
+
+  if [ ! -f "$update_mod_sh_path" ]; then
+    printf "Unable to find ${update_mod_sh_path}\nExiting...\n"
+    exit 1
+  fi
+
+  "$update_mod_sh_path" "$mod_version"
+}
+
+initialize_upgrade_module() {
+  if [ -z "$UPGRADE_DIR" ] || [ -z "$MOD_VERSION" ]; then
+    printf "UPGRADE_DIR and MOD_VERSION vars need to be set.\nExiting...\n"
+    exit 1
+  fi
+
+  printf "Creating Upgrade deploy DIR..."
+  mkdir -p "$UPGRADE_DIR"
+
+  printf "Initializing module on %s\n" "$UPGRADE_DIR"
+  terraform -chdir="$UPGRADE_DIR" init -backend=false -from-module="github.com/dominodatalab/terraform-aws-eks.git//examples/deploy?ref=${MOD_VERSION}"
+
+  printf "Updating module source...\n"
+  update_mod_source "$UPGRADE_DIR" "$MOD_VERSION"
+}
+
+backup_files() {
+  if [ ! -d "$BACKUP_DIR" ] || [ -z "$(ls -A "$BACKUP_DIR")" ]; then
+    printf "Backing up %s to %s.\n" "$DEPLOY_DIR" "$BACKUP_DIR"
+    cp -r "$DEPLOY_DIR" "$BACKUP_DIR"
+  else
+    printf "%s exists and is not empty. Skipping backup...\n" "$BACKUP_DIR"
+  fi
+}
+
+copy_updated_files() {
+  backup_files
+  rsync -av --include={'*.tf','*.sh'} --exclude={'.*/','*.tfvars'} "${UPGRADE_DIR}/" "${DEPLOY_DIR}/"
+}
+
+is_vpc_cni_installed() {
+  local deploy_id=$(hcledit attribute get deploy_id -f "$INFRA_VARS" | jq -r .)
+
+  if aws eks list-addons --cluster-name "${deploy_id}" --output json | jq -e '.addons | index("vpc-cni")' >/dev/null; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+update_tfvars_from_module() {
+  declare -A tfvars_paths=(
+    ["$INFRA_DIR"]=$INFRA_VARS
+    ["$CLUSTER_DIR"]="$CLUSTER_VARS"
+    ["$NODES_DIR"]="$NODES_VARS"
+  )
+
+  for mod_dir in "${!tfvars_paths[@]}"; do
+    vars_path="${tfvars_paths[$mod_dir]}"
+    if [ "$(basename $mod_dir)" = "cluster" ] && is_vpc_cni_installed; then
+      echo
```

Comment:
Is this a placeholder for adding to the addon list or something?

Reply:
Initially I thought to make it smart but it is not trivial to edit the hcl nested value. I want to just print a message but forgot to finish the line.